### ISSUE TITLE
Request for new feature: custom task(job)

### ISSUE THREAD

**Issue by korabelnikov** [NVIDIA/DIGITS#1175]
I noticed a lack of some custom task (besides classification, segmentation, objdetecetion, creating db). I think it will be useful if you do some preprocess or postproces things.
In my case it is extracting from volume image and augmentation as preprocess, and validation on full image.

I saw,that NVidia Digits(c) Roadmap have a paragraph with something similar about version 2
![image](https://cloud.githubusercontent.com/assets/3808833/19470352/36d9dada-9530-11e6-99bb-87521f99eb8f.png)

`more job management options.`
maybe it's relevant

**Comment by gheinrich** [comment]
> I think it will be useful if you do some preprocess or postproces things.

In DIGITS 5.0 we support plug-ins for data ingestion and inference visualization. I suppose this addresses what you mean. You are free to implement your own plug-ins to extend DIGITS if your use case isn't natively supported. We have yet to write documentation for the plug-in API as we're waiting for initial feedback before doing so. Example plug-ins may be found in the [plug-in folder](https://github.com/NVIDIA/DIGITS/tree/digits-5.0/plugins).

**Comment by korabelnikov** [comment]
Yes, it's helpfull answer. Plugin-based architecture is very good. But what you think about to add such feature to core of DIGITS?
Maybe i can implement it.

**Comment by gheinrich** [comment]
Thanks for the offer! Can you explain in more detail the feature you're interested in? If it's not something the majority of users would need it's better to distribute it as a plug-in. It's easy to convert a plug-in into a built-in extension, if we ever feel the need to do it.

**Comment by korabelnikov** [comment]
@gheinrich , it's simple command job, wich is on a par with 'create database' and 'create model' jobs.
The result of command is return code, that indicate success or failure.
User can simply performs his own command in digits pipeline.

Widely used example:
Everyone of us starts with downloading  data. We download it, and time by time we look on the process in order to start next job with minimal idle of server.
next stage may be some preprocessing of input data. 
Then creating db, and learning model.

Custom command may have only input text field for user's input. For example 
`prepare_my_data.sh /root/mydata --param1 --param2`
if it terminated with non-zero ret code, then next job doesn't starts

What you think about it?

**Comment by gheinrich** [comment]
Hi @korabelnikov thanks for the explanation I had misunderstood your point originally. I think it's a good idea! If you're willing to help implement this we can write a detailed spec together and get you started on this.

**Comment by korabelnikov** [comment]
@gheinrich very good.
let's talk about spec.
